File: Assignment_44X/Assignment_44.1.cpp
#include<iostream>
using namespace std;

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;

class SinglyLL
{
    private:
        PNODE first;
        int iCount;

    public:

        SinglyLL()
        {
            cout<<"Object of SinglyLL gets created"<<"\n";
            first = NULL;
            iCount = 0;
        }

    void InsertFirst(int no)
    {
        PNODE newn = NULL;

        newn = new NODE;

        newn->data = no;
        newn->next = NULL;

        if(first == NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
    }

    void Display()
    {
        PNODE temp = NULL;

        temp = first;

        while(temp != NULL)
        {
            cout<<"| "<<temp->data<<" |->";
            temp = temp->next;
        }

        cout<<"NULL\n";
    }

    bool Search(int iNo)
    {
        PNODE temp = NULL;

        temp = first;

        while(temp != NULL)
        {
            if(temp->data == iNo)
            {
                return true;
                temp = temp->next;
            }
            temp = temp->next;
            
        }
        return false;
    }
    
};

int main()
{
    SinglyLL obj;
    bool bRet = false;

    obj.InsertFirst(51);
    obj.InsertFirst(41);
    obj.InsertFirst(31);
    obj.InsertFirst(21);
    obj.InsertFirst(11);

    obj.Display();

    bRet = obj.Search(41);

    if(bRet == true)
    {
        cout<<"Elements is Present\n";
    }
    else
    {
        cout<<"Elements is Not Present\n";
    }

    return 0;
}

File: Assignment_44XX/Assignment_44_1.java
class node
{
    public int data;
    public node next;
    
    public node(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private node first;
    private int iCount;

    public SinglyLL() 
    {
        System.out.println("Object Of SinglyLL gets created");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {
        node newn  = null;

        newn = new node(no);

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
    }
    public void Display()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" |->");
            temp = temp.next;
        }

        System.out.println("null\n");
    }

    public boolean Search(int iNo)
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            if(temp.data == iNo)
            {
                return true;
            }
            temp = temp.next;
        }

        return false;

    }

    
}

class Assignment_44_1
{
    public static void main(String Arr[])
    {
        SinglyLL obj;

        boolean bRet = false;

        obj = new SinglyLL();

        obj.InsertFirst(51);
        obj.InsertFirst(41);
        obj.InsertFirst(31);
        obj.InsertFirst(21);
        obj.InsertFirst(11);

        obj.Display();

        bRet = obj.Search(31);

        if(bRet == true)
        {
            System.out.println("Element is Found");
        }
        else
        {
            System.out.println("Element is Not found");
        }
        
        obj = null;

        System.gc();
    }
}